server/src/pages/shotindex/view.js

```diff
@@ -42,7 +41,7 @@ class Body extends React.Component {
         <div className="column-space full-height default-color-scheme" id="shot-index-page">
           <div id="shot-index-header" className="header">
             <h1><a href="/shots">Firefox <strong>Screenshots</strong> <sup>Beta</sup></a></h1>
-            {this.props.enableUserSettings ? this.renderSettingsPage() : null}
+            {this.props.enableUserSettings && this.props.hasDeviceId ? this.renderFxASignIn() : null}
```

Comment:
After signing in from My Shots, I get teh following mark up for the Setting link:

![image](https://user-images.githubusercontent.com/198055/44929135-86164d00-ad1f-11e8-9720-b27eecfe8586.png)

Note that's "Settings" in the Sign In anchor.  And it was not clickable.  I think `SignInButton` probably need to (re)render based on its state, not props.

Reply:
@chenba Updated PR with fix of this issue by checking isAuthenticated prop inside click handler.  
As suggested, using state to force re-render doesn't seem to help here even after state is assigned to updated prop value inside getDerivedStateFromProps